src/LibraryUtilities.ts

```diff
@@ -250,6 +286,34 @@ export function constructLibraryItem(
 }
 
 /**
+ * Compare two arrays of strings and check if the strings in the first array match
+ * those in the second array.
+ * 
+ * @param {string[]} parts 
```

Comment:
Can we keep the names as `parts` and `partsToInclude`, since the two parameters are not interchangeable?
This is a method to replace `matchUpTo`, to check if `parts` contains all strings in `partsToInclude` (and if `parts[0] == partsToInclude[0]`,  `parts[1] == partsToInclude[1]` and so on). I've added some examples into the comments.